src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    [EventSource(
+        Guid = "20b30044-b729-457e-8dda-8b41b5dd02e6", 
+        Name = "System.Buffers.BufferPoolEventSource",
+        LocalizationResources = "FxResources.System.Buffers.SR")]
+    internal sealed class ArrayPoolEventSource : EventSource
+    {
+        internal readonly static ArrayPoolEventSource Log = new ArrayPoolEventSource();
+
+        private const EventTask Buffer = (EventTask)1;
+
+        [Event(1, Level = EventLevel.Informational, Opcode = EventOpcode.Start, Task = Buffer)]
+        internal void BufferRented(int bufferId, int bufferSize, int poolId, int bucketId) { WriteEvent(1, bufferId, bufferSize, poolId, bucketId); }
+
+        [Event(2, Level = EventLevel.Informational, Opcode = EventOpcode.Stop, Task = Buffer)]
+        internal void BufferReturned(int bufferId, int poolId) { WriteEvent(2, bufferId, poolId); }
+
+        [Event(3, Level = EventLevel.Informational)]
+        internal void PooledBufferCreated(int bufferId, int bufferSize, int poolId, int bucketId) { WriteEvent(3, bufferId, bufferSize, poolId, bucketId); }
+
+        [Event(4, Level = EventLevel.Warning, Opcode = EventOpcode.Start, Task = Buffer)]
+        internal void UnPooledBufferAllocated(int bufferId, int bufferSize, int poolId) { WriteEvent(4, bufferId, bufferSize, poolId); }
+
+        [Event(5, Level = EventLevel.Warning, Opcode = EventOpcode.Start, Task = Buffer)]
+        internal void BufferOverMaximumAllocated(int bufferId, int bufferSize, int maximumSize, int poolId) { WriteEvent(5, bufferId, bufferSize, maximumSize, poolId); }
+
+        [Event(6, Level = EventLevel.Warning)]
+        internal void BucketExhausted(int bucketId, int bucketSize, int buffersInBucket, int poolId) { WriteEvent(6, bucketId, buffersInBucket, bucketSize, poolId); }
```

Comment:
The arguments to the method should be the same as they are to WriteEvent.  You can reorder the fills in the text message accordingly.